FormidableLabs/react-native-app-auth
# add configurable timeout fn param (connectionTimeoutSeconds/connectionTimeoutMillis)

Fixes #236 #479

## Description

- adds a connectionTimeoutSeconds fn param to prefetchConfiguration, authorize and refresh
- adds connectionTimeoutSeconds property to AuthConfiguration type
- adds connectionTimeoutSeconds param validation and test cases
- converts value to milliseconds for Android platform
- adds logic to native methods to handle new fn param
- adds android specific test cases for time conversion (seconds -> milliseconds)

Random addition
- adds default value for `warmAndPrefetchChrome` fn param (fixes 479)

## Steps to verify

(not sure if this is the best way to go about it 🤷‍♂️ )

1. Manually sync the Example/node_modules/react-native-app-auth directory with the updated ios/android dirs, react-native-app-auth.podspec and index.js files.

`cp -r ios Example/node_modules/react-native-app-auth && cp -r android Example/node_modules/react-native-app-auth && cp index.js Example/node_modules/react-native-app-auth && cp react-native-app-auth.podspec Example/node_modules/react-native-app-auth`

2. change the `issuer` property of the identityserver config in Example/App.js to a url that will never resolve (ex. http://www.example:81.com)

3. If the url doesn't use https protocol, add `dangerouslyAllowInsecureHttpRequests: true`  to the argument object passed to the 'authorize' method in Example/App.js.

4. Run the Example app and attempt to authorize through identity server. The default timeout is 60 seconds for iOS and 15 seconds for Android. It should be apparent that the configured timeout of 5 seconds is being respected.
